dataland-email-service/src/main/kotlin/org/dataland/datalandemailservice/services/EmailUnsubscriber.kt

```diff
@@ -0,0 +1,76 @@
+package org.dataland.datalandemailservice.services
+
+import org.dataland.datalandemailservice.email.EmailSender
+import org.dataland.datalandemailservice.repositories.EmailSubscriptionRepository
+import org.dataland.datalandmessagequeueutils.messages.InternalEmailMessage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+/**
+ * Service responsible for managing email subscriptions.
+ */
+@Component
+class EmailUnsubscriber(
+    @Autowired private val emailSubscriptionRepository: EmailSubscriptionRepository,
+    @Autowired private val emailSender: EmailSender,
+    @Autowired private val internalEmailBuilder: InternalEmailBuilder,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Unsubscribes an email address from email communications based on the provided UUID and sends a mail
+     * to the stakeholders that this email address is now unsubscribed.
+     * @param uuid The UUID of the subscription to be unsubscribed.
+     * @return A response entity indicating the result of the unsubscription.
+     */
+    @Transactional
+    fun unsubscribeUuidAndSendMailToStakeholders(uuid: UUID): ResponseEntity<String> {
+        val emailSubscription = emailSubscriptionRepository.findByUuid(uuid)
+
+        return if (emailSubscription != null) {
+            unsubscribeEmailWithUuid(emailSubscription.uuid)
+            sendUnsubscriptionEmail(emailSubscription.emailAddress)
```

Comment:
This looks like the unsubscription mail is sent only to the formerly subscribed email and not to the stakeholders

Reply:
changed to informStakeholdersOfUnsubscription